design-proposals/klip-63-pause-resume-persistent-queries.md

```diff
@@ -0,0 +1,66 @@
+# KLIP-63: PAUSE and RESUME for persistent queries 
+
+**Author**: Jim Hughes (jnh5y) | 
+**Release Target**: Unknown | 
+**Status**: In Discussion | 
+**Discussion**: 
+
+**tl;dr:** Adds the ability to pause the processing of a persistent query.
+           
+## Motivation and background
+
+Users may wish to pause a query for a number of reasons and then resume processing.  As examples, users may wish to modify data pipelines built from persistent queries or to manage computational resources.
```

Comment:
Just curious - since the size of the ksql cluster is maintained, users wouldn't see any cost savings from this, would they? Or would they see some since the queries wouldn't be processing?

Reply:
Presently, they would not see costs savings from pausing individual queries.

I was meaning "manage computational resources" to suggest that some queries may process a little faster.  

If one was counting pennies, maybe one could find some loose change in savings for how long data was in Kafka?  Overall, this is not a feature aimed at reducing costs.

Reply:
That makes sense, I was just making sure my understanding was correct 😄  Maybe worth making sure that the customer knows that their bill will stay the same

Reply:
Hmm...  I don't think we need to provide billing details in this KLIP.  (Shout if I'm wrong!)